src/herder/TransactionQueue.cpp

```diff
@@ -0,0 +1,228 @@
+    {
+        return TransactionSubmitStatus::ERROR;
+    }
+
+    auto sourceAccount = stellar::loadAccount(ltx, tx->getSourceID());
+    if (getAvailableBalance(ltx.loadHeader(), sourceAccount) <
+        state.mTotalFees + tx->getFeeBid())
+    {
+        tx->getResult().result.code(txINSUFFICIENT_BALANCE);
+        return TransactionSubmitStatus::ERROR;
+    }
+
+    auto map = findOrAdd(mPendingTransactions[0], tx->getSourceID());
+    map->addTx(tx);
+
+    return TransactionSubmitStatus::PENDING;
+}
+
+void
+TransactionQueue::remove(std::vector<TransactionFramePtr> const& dropTxs)
+{
+    for (auto& m : mPendingTransactions)
+    {
+        if (m.empty())
+        {
+            continue;
+        }
+
+        std::set<std::shared_ptr<TxMap>> toRecalculate;
+
+        for (auto const& tx : dropTxs)
+        {
+            auto const& acc = tx->getSourceID();
+            auto const& txID = tx->getFullHash();
+            auto i = m.find(acc);
+            if (i != m.end())
+            {
+                auto& txs = i->second->mTransactions;
+                auto j = txs.find(txID);
+                if (j != txs.end())
+                {
+                    txs.erase(j);
+                    if (txs.empty())
+                    {
+                        m.erase(i);
+                    }
+                    else
+                    {
+                        toRecalculate.insert(i->second);
+                    }
+                }
+            }
+        }
+
+        for (auto txm : toRecalculate)
+        {
+            txm->recalculate();
+        }
+    }
+}
+
+bool
+TransactionQueue::contains(TransactionFramePtr tx) const
+{
+    return std::any_of(std::begin(mPendingTransactions),
+                       std::end(mPendingTransactions),
+                       [&](AccountTxMap const& map) {
+                           auto txMap = map.find(tx->getSourceID());
+                           if (txMap == map.end())
+                           {
+                               return false;
+                           }
+
+                           auto& transactions = txMap->second->mTransactions;
+                           return transactions.find(tx->getFullHash()) !=
+                                  std::end(transactions);
+                       });
+}
+
+AccountTransactionsState
+TransactionQueue::getState(AccountID const& accountID) const
+{
+    AccountTransactionsState state;
+
+    for (auto& map : mPendingTransactions)
+    {
+        auto i = map.find(accountID);
+        if (i != map.end())
+        {
+            auto& txmap = i->second;
+            state.mTotalFees += txmap->mCurrentState.mTotalFees;
+            state.mMaxSeq =
+                std::max(state.mMaxSeq, txmap->mCurrentState.mMaxSeq);
+        }
+    }
+
+    return state;
+}
+
+void
+TransactionQueue::shift()
+{
+    mPendingTransactions.erase(--mPendingTransactions.end());
+    mPendingTransactions.emplace_front();
+}
+
+int
```

Comment:
inconsistent type: int vs uint64

Reply:
you didn't fix this one: this emits a warning as size_t -> int is unsafe, just use static_cast